src/screens/Filter/components/FilterItem/useFilterItemController.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback } from 'react';
+
+import { FilterItemValueModel } from 'src/models';
+
+export interface UseFilterItemControllerParams {
+  filterTitle: string;
+  filterName: string;
+  values: Array<FilterItemValueModel>;
+  multiple: boolean;
+  onFilterChange: (filterName: string, value: string) => void;
+}
+
+export const useFilterItemController = (params: UseFilterItemControllerParams) => {
+
+  const onMultipleFilterChange = useCallback((value: string, previousValueString: string) => {
+    if (previousValueString.includes(value)) {
+
+      return params.onFilterChange(params.filterName, previousValueString.split(',')
+        .filter((item) => item !== value).toString().replaceAll(',', '|'));
+    }
+
+    const currentValuesString = previousValueString ? previousValueString.replaceAll(',', '|') + `|${value}` : value;
+
+    return params.onFilterChange(params.filterName, currentValuesString);
+  }, [params.multiple]);
+
+  const handleChange = useCallback((value: string) => {
+    const previousValueString = params.values.filter((v) => v.isActive === true)
+      .map((item) => item.value)
```

Comment:
the best way to handle filters in your case is to create a dedicated manager.
It could look like(this is pseudo-code)
```
FiltersValuesManager {
  function getRawValues(queryString) -> returns an array of filter values based on the string, e.g "filterValue1|filterValue2" transforms to ["filterValue1", "filterValue2"], or "filterValue" to ["filterValue"]
  
  function getQueryValues(rawValues) -> returns a string of filter values based on the raw values array, e.g ["filterValue1", "filterValue2"] to "filterValue1|filterValue2", or ["filterValue"] to "filterValue"


function getAppliedFiltersString(baseString, nextRawValueString, multiple) -> {
   const rawValues = getRawValues(baseString)
   
   checks if the value already applied
   if (rawValues.includes(nextRawValueString)) {
      that means we need to turn the filter off, therefore we need to remove this value from the filter and returns queryString
       return getQueryValues(rawValues.filter((value) => value === nextRawValueString))
   }
   
   otherwise we consider multiple boolean flag in order to properly apply the filter
   
   if (multiple) {
     return getQueryValues([...rawValues, nextRawValueString])
   }
   
   otherwise just return single query value
   return getQueryValues([nextRawValueString])
 }
}
```